client/quantum_serverless/core/program.py

```diff
@@ -72,21 +74,58 @@ class Program:  # pylint: disable=too-many-instance-attributes
 
     @classmethod
     def from_json(cls, data: Dict[str, Any]):
-        """Reconstructs Program from dictionary."""
-        field_names = set(f.name for f in dataclasses.fields(Program))
-        return Program(**{k: v for k, v in data.items() if k in field_names})
+        """Reconstructs QiskitPattern from dictionary."""
+        field_names = set(f.name for f in dataclasses.fields(QiskitPattern))
+        return QiskitPattern(**{k: v for k, v in data.items() if k in field_names})
 
     def __str__(self):
-        return f"Program({self.title})"
+        return f"QiskitPattern({self.title})"
 
     def __repr__(self):
         return self.__str__()
 
 
+class Program(QiskitPattern):
+    """[Deprecated] Program."""
+
+    def __init__(
+        self,
+        title: str,
+        entrypoint: Optional[str] = None,
+        working_dir: Optional[str] = "./",
+        env_vars: Optional[Dict[str, str]] = None,
+        dependencies: Optional[List[str]] = None,
+        description: Optional[str] = None,
+        version: Optional[str] = None,
+        tags: Optional[List[str]] = None,
+        raw_data: Optional[Dict[str, Any]] = None,
+    ):
+        """[Deprecate] Serverless program."""
+        warnings.warn(
+            "`Program` class has been deprecated "
+            "and will be removed in future releases. "
+            "Please, use `QiskitPattern` instead."
+        )
+        super().__init__(
+            title,
+            entrypoint,
+            working_dir,
+            env_vars,
+            dependencies,
+            description,
+            version,
+            tags,
+            raw_data,
+        )
+
+    def __repr__(self):
+        return f"Program({self.title})"
+
+
 class ProgramStorage(ABC):
```

Comment:
Do we need to `s/Program/Pattern/` here (and elsewhere in this class, i.e. `save_program` and the like)?
Or :shudder: change the filename itself from program to pattern?

Reply:
good point!